### DISCUSSION TITLE
Updating with zsync

### DISCUSSION THREAD

**Discussion by dekrain** [ppy/osu#16908] [Q&A] [answered]
While I was investigating the bug that later got fixed in #16903, to see if it was my local build failing, I wanted to check the stable client.
I had the previous build AppImage (2022.205.0) and wanted to update to 2022.216.0 using zsync, but it was not working. I downloaded the zsync file, because zsync doesn't work over HTTPS, and got an error saying `failed to retrieve from osu.AppImage`. So my question is:

Has anyone been able to do the update over zsync? If so, it would be appreciated if there were instructions on how to do it somewhere, like in the README or on the wiki.

**Comment by dekrain** [comment, marked as answer]
After some digging, I found [this](https://docs.appimage.org/packaging-guide/optional/updates.html#id6) guide on updating an AppImage-based application. It says that the application may bundle with either `AppImageUpdate`/`appimageupdatetool` or a `libappimageupdate`-based solution. It may be worth integrating something like this into the game as an alternative to Squirrel on Windows.

**Reply by peppy** [reply]
The squirrel fork we use is actually looking to add linux support, so we'll wait to see what happens there first.
